jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java

```diff
@@ -25,30 +26,60 @@
 @RequiredArgsConstructor
 public class WebScreenshotCreator {
 
-    public static final String SCREENSHOTS_KEY = "error_screenshots";
-    private static final String FILE_NAME_PATTERN = "FAILED_%s.png";
+    public enum Type {
+        FAILED(0), DEBUG(3), MANUAL(0), STEP(2), WAIT(1);
+        private final int hierarchy;
+
+        Type(int hierarchy){
+            this.hierarchy = hierarchy;
+        }
+
+        public int getHierarchy(){
+            return hierarchy;
+        }
+    }
+
+    public static final String FAILED_SCREENSHOTS_KEY = "error_screenshots";
+    public static final String STEP_SCREENSHOTS_KEY = "step_screenshots";
```

Comment:
Its not always step screenshots since it can be manual, wait etc.... How about something like report_screenshots?

Reply:
done